### PR TITLE
Delayed Spawn fix

### PR DESCRIPTION
**Pull request by elgthedev**
- add coroutine to spawn enemies and allies
- fix allies list
- add flag teleportTriggered for compatibility with other teleport mods

### CODE REVIEW

File: TeleportEverything/HarmonyPatches.cs [hunk 1]
```diff
@@ -194,9 +199,29 @@ public class TeleportTo_Patch
...
+        public static DelayedAction delayedAction;
+        [HarmonyPatch(typeof(Player), nameof(Player.Awake))]
+        public class PlayerAwake_Patch
+        {
+            static void Postfix(Player __instance)
+            {
+                PlayerController controller = __instance.GetComponent<PlayerController>();
+                if (delayedAction == null)
+                {
+                    delayedAction = controller.gameObject.AddComponent<DelayedAction>();
```
Comment (KPro1982) [inline TeleportEverything/HarmonyPatches.cs:223]: What's the purpose of using the playercontroller object here?

File: TeleportEverything/DelayedAction.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace TeleportEverything
+{
+    public class DelayedAction:MonoBehaviour
+    {
+        public void InvokeDelayed(System.Action aDelegate, float delay)
```
Comment (KPro1982) [inline TeleportEverything/DelayedAction.cs:13]: I agree, coroutine is better here. Interestingly, the game uses a loop timer in this situation.

Comment (elgthedev) [reply TeleportEverything/DelayedAction.cs:13]: btw I think we can test if the spawn time will be the same for both players.

File: TeleportEverything/HarmonyPatches.cs (hunk 1 above)
Comment (elgthedev) [reply TeleportEverything/HarmonyPatches.cs:223]: actually I added the component DelayedAction to PlayerController just to instance the DelayedAction object as it inherits from MonoBehaviour

Comment (elgthedev) [reply TeleportEverything/HarmonyPatches.cs]: Ohh now I see what you are talking about, I could've attached it directly into the player obj. I just refactored it to insert delayed action into the game object. Let me know what you think.

Comment (KPro1982) [reply TeleportEverything/HarmonyPatches.cs]: Got it.

**Pull request merged**